### PR TITLE
Fix beat sync stopping after returning to menu from a failed play

### LINKED ISSUE #20193: Main menu icons no longer animate after failing a song
Can repro on 9aab502adcc6e61bb675eafcfaf5368f828e1ab8

### Discussed in https://github.com/ppy/osu/discussions/20192

<div type='discussions-op-text'>

<sup>Originally posted by **TheEggo** September  8, 2022</sup>
When exiting a song, objects that would usually pulse to the bpm seem to just be stuck (e.g osu! logo, menu icons etc)

https://user-images.githubusercontent.com/64386789/189114176-a47717d8-0842-4fef-8c7a-b51095cd5b5a.mp4

</div>

### PR DESCRIPTION
**Pull request by peppy**
Closes #20193.

Explanation is inline comment.

### CODE REVIEW

File: osu.Game/Beatmaps/FramedBeatmapClock.cs [hunk 1]
```diff
@@ -121,6 +121,17 @@ protected override void LoadComplete()
         protected override void Update()
         {
             base.Update();
+
+            if (Source != null && Source is not IAdjustableClock && Source.CurrentTime < finalClockSource.CurrentTime)
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/FramedBeatmapClock.cs:125]: Can you assure me that this isn't going to have some weird unexpected effects on gameplay?

Comment (smoogipoo) [reply osu.Game/Beatmaps/FramedBeatmapClock.cs:125]: Actually, just in general, I don't know how to reassure myself that this isn't going to cause some sort of catastrophic failure. It's called every frame even in song select when time is progressing normally. The track is practically always behind the final clock time isn't it?

Comment (peppy) [reply osu.Game/Beatmaps/FramedBeatmapClock.cs:125]: In all cases but `OsuGameBase`, the source is `IAdjustableClock` (ie. a track) which will never run this code.

Comment (peppy) [reply osu.Game/Beatmaps/FramedBeatmapClock.cs:125]: You raise a good point though, this should be checking `decoupledClock` not `finalClockSource` (as offsets should not be involved).

Of note, I tested before PRing this and nothing breaks even if this is called every frame.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**